packages/adminrouter/extra/src/docs/api/nginx.master.yaml

```diff
@@ -0,0 +1,332 @@
+          $oauth_auth_redirector$uri?client=$oauth_client_id&cluster_id=$cluster_id
+        type: redirect
+    path: /login
+  /capabilities:
+    group: Capabilities
+    matcher: path
+    description: List of capabilities supported by DC/OS
+    proxy:
+      path: 'http://$backend/capabilities'
+      backend: cosmos
+    path: /capabilities
+  /cosmos/service/:
+    group: Cosmos
+    matcher: path
+    description: Start a DC/OS service from a DC/OS package
+    proxy:
+      path: 'http://$backend/service/'
+      backend: cosmos
+    path: /cosmos/service/
+  /exhibitor:
+    group: Exhibitor
+    matcher: exact
+    description: Redirect to add trailing slash
+    visibility: hidden
+    redirects:
+      - regex: ^/exhibitor$
+        replacement: '$scheme://$http_host/exhibitor/'
+        type: permanent
+    path: /exhibitor
+  /exhibitor/:
+    group: Exhibitor
+    matcher: path
+    description: Manage Zookeeper
+    proxy:
+      path: 'http://$backend/'
+      backend: exhibitor
+    path: /exhibitor/
+  /dcos-history-service/:
+    group: History
+    matcher: path
+    proxy:
+      path: 'http://$backend/'
+      backend: dcos_history_service
+    path: /dcos-history-service/
+  /marathon:
+    group: Marathon
+    matcher: exact
+    description: Redirect to add trailing slash
+    visibility: hidden
+    redirects:
+      - regex: ^/marathon$
+        replacement: '$scheme://$http_host/marathon/'
+        type: permanent
+    path: /marathon
+  /marathon/:
+    group: Marathon
+    matcher: path
+    proxy:
+      path: 'http://$backend/'
+      backend: marathon
+    path: /marathon/
+  /cache/master/:
+    group: Mesos
+    matcher: path
+    description: Cache of the Mesos master API
+    proxy:
+      path: 'http://$backend/master/'
+      backend: mesos
+    path: /cache/master/
+  /mesos:
+    group: Mesos
+    matcher: exact
+    description: Redirect to add trailing slash
+    visibility: hidden
+    redirects:
+      - regex: ^/mesos$
+        replacement: '$scheme://$http_host/mesos/'
+        type: permanent
+    path: /mesos
+  /mesos/:
+    group: Mesos
+    matcher: path
+    description: Apache Mesos
+    proxy:
+      path: 'http://$backend/'
+      backend: mesos
+    path: /mesos/
+  /mesos_dns:
+    group: Mesos DNS
+    matcher: exact
+    description: Redirect to add trailing slash
+    visibility: hidden
+    redirects:
+      - regex: ^/mesos_dns$
+        replacement: '$scheme://$http_host/mesos_dns/'
+        type: permanent
+    path: /mesos_dns
+  /mesos_dns/:
+    group: Mesos DNS
+    matcher: path
+    proxy:
+      path: 'http://$backend/'
+      backend: mesos_dns
+    path: /mesos_dns/
+  /dcos-metadata/:
+    group: Metadata
+    matcher: path
+    description: DC/OS metadata
+    file: /opt/mesosphere/active/dcos-metadata/etc/
+    path: /dcos-metadata/
+  /dcos-metadata/dcos-version.json:
+    group: Metadata
+    matcher: path
+    description: DC/OS version (unauthenticated)
```

Comment:
This is still up for debate... 
I haven't specifically stated that everything is authenticated unless specified, but it seems silly to mark everything else as authenticated. Because authentication is enforced using a custom Lua script, there's no way to know from a generic NGINX perspective what that Lua script does. So it doesn't make sense for Ngindox (which is not dc/os specific) to know that...

So I'm not really sure how to programmatically determine auth-required status per route.

That's why I just specified it via author-defined description string.